### ISSUE TITLE
...some screens get skipped mysteriously

### ISSUE THREAD

**Issue by agnescameron** [Kallirroi/bot_test#76]
where do they go? who ate the screens? do they simply appear on someone else's computer? etc

**Comment by agnescameron** [comment]
probably related to https://github.com/Kallirroi/bot_or_not/issues/75

**Comment by agnescameron** [comment]
this is now fixed.

The problem that was solved by #75 was that the timer wasn't clearing properly at the start of every round. So, if you took your time (for example) entering your name or starting a chat, the timer would already be 'over' by the time you went to the next screen. This got solved by making the setting of `this.shouldUpdate` only happen *after* the timer had been reset, using:

```
    this.setState({ timerStart: Date.now()},
      () => this.setState({ timerTime: 0},
      () => this.shouldUpdate = true))
```
...which worked really great, apart from:

*now* there were only 2 screens getting skipped, and only after the `checkTimeout` screens. What was happening was that, while the callback was happening, the loop had already looped around again and was getting called twice, effectively skipping a screen.

This is solved by just `await`-ing `checkTimeout` in `configureState`, which seems to work pretty well!
